### PR TITLE
[WIP] Repo.Listing changes

### PR DESCRIPTION
**Pull request by amangale**
This PR builds upon https://github.com/OpenBazaar/openbazaar-go/pull/1666. 
This introduces repo.Listing which will be used to handle different versions of existing/legacy contracts in the nodes. Repo.Listing will used for interaction between pb.Listing and the core.

### CODE REVIEW

File: repo/db/cases.go [hunk 1]
```diff
@@ -254,7 +254,11 @@ func (c *CasesDB) GetAll(stateFilter []pb.OrderState, searchTerm string, sortByA
 		contract := new(pb.RicardianContract)
 		err := jsonpb.UnmarshalString(string(buyerContract), contract)
 		if err != nil {
-			jsonpb.UnmarshalString(string(vendorContract), contract)
+			err = jsonpb.UnmarshalString(string(vendorContract), contract)
```
Comment (placer14) [inline repo/db/cases.go]: Intended use of `jsonb.UnmarshalString`?

File: repo/db/cases.go [hunk 2]
```diff
@@ -254,7 +254,11 @@ func (c *CasesDB) GetAll(stateFilter []pb.OrderState, searchTerm string, sortByA
...
+			if err != nil {
+				//return ret, 0, err
+				log.Error(err)
```
Comment (placer14) [inline repo/db/cases.go]: Everything below relies on the successful unmarshaling of the contract. Any reason against continuing past this iteration which failed?

File: repo/db/followers_test.go [hunk 3]
```diff
@@ -102,15 +120,24 @@ func TestDeleteFollower(t *testing.T) {
...
+		t.Error(err)
+	}
 	err = fdb.Delete("abc")
 	if err != nil {
 		t.Error(err)
 	}
-	stmt, _ := fdb.PrepareQuery("select peerID from followers where peerID=?")
+	stmt, err := fdb.PrepareQuery("select peerID from followers where peerID=?")
+	if err != nil {
+		t.Log(err)
```
Comment (placer14) [inline repo/db/followers_test.go]: Setup steps of a test should Fatally fail if the rest of the checks have no chance of succeeding.

File: schema/constants.go [hunk 4]
```diff
@@ -95,16 +95,16 @@ const (
...
 	CoinAPIOpenBazaarETH = "https://mainnet.infura.io"
 
-	CoinAPIOpenBazaarTBTC = "https://tbtc.api.openbazaar.org/api"
-	CoinAPIOpenBazaarTBCH = "https://tbch.api.openbazaar.org/api"
-	CoinAPIOpenBazaarTLTC = "https://tltc.api.openbazaar.org/api"
-	CoinAPIOpenBazaarTZEC = "https://tzec.api.openbazaar.org/api"
+	CoinAPIOpenBazaarTBTC = "https://tbtc.blockbook.api.openbazaar.org/api"
+	CoinAPIOpenBazaarTBCH = "https://tbch.blockbook.api.openbazaar.org/api"
+	CoinAPIOpenBazaarTLTC = "https://tltc.blockbook.api.openbazaar.org/api"
+	CoinAPIOpenBazaarTZEC = "https://tzec.blockbook.api.openbazaar.org/api"
```
Comment (placer14) [inline schema/constants.go]: I don't think we're changing these back.

File: core/listings.go [hunk 5]
```diff
@@ -311,63 +151,60 @@ func (n *OpenBazaarNode) SetListingInventory(listing *pb.Listing) error {
 }
 
 // CreateListing - add a listing
-func (n *OpenBazaarNode) CreateListing(listing *pb.Listing) error {
-	exists, err := n.listingExists(listing.Slug)
+func (n *OpenBazaarNode) CreateListing(r []byte) (string, error) {
+	listing, err := repo.CreateListing(r, n.TestNetworkEnabled() || n.RegressionNetworkEnabled(), &n.Datastore, n.RepoPath)
```
[the remark this answers is not in the document]
Comment (cpacia) [reply core/listings.go]: agreed

File: repo/listing.go [hunk 6]
```diff
@@ -0,0 +1,2451 @@
...
+	out, err := m.MarshalToString(l)
+	if err != nil {
+		return nil, err
+	}
+
+	out1, err := json.Marshal(out)
+	if err != nil {
+		return nil, err
+	}
+	log.Info(len(out1))
```
[the remark this answers is not in the document]
Comment (cpacia) [reply repo/listing.go]: yes out and out 1 look like they are not needed.

**Comment by placer14** [comment]
As discussed, we will resolve these outstanding remarks as separate issues. Thank you for extracting the issues and getting this merged, @cpacia. (And, of course, thanks for the effort on these commits, @amangale.)

**Pull request merged**